src/rust/lib/graph-editor/src/component/node/port/output.rs

```diff
@@ -237,63 +241,131 @@ impl OutputPorts {
+                eval_ view.events.mouse_down(frp.on_port_mouse_down.emit(index));
+
+
+                 // === Animation Handling == ///
+
                  eval port_size.value    ((size) shape.grow.set(*size));
                  eval port_opacity.value ((size) shape.opacity.set(*size));
 
-                is_resize_target <- set_port_size.map(move |(id,_)| *id == index);
-                size_change      <- set_port_size.gate(&is_resize_target);
-                eval size_change (((_, size)) port_size.set_target_value(*size));
 
-                is_opacity_target <- set_port_opacity.map(move |(id, _)| *id==index);
-                opacity_change    <- set_port_opacity.gate(&is_opacity_target);
-                eval opacity_change (((_, opacity)) port_opacity.set_target_value(*opacity));
+                // === Visibility and Highlight Handling == ///
 
-                eval_ view.events.mouse_over ([port_size,set_port_sizes_all,port_opacity,
-                                              set_port_opacity_all] {
-                    set_port_sizes_all.emit(BASE_SIZE);
-                    set_port_opacity_all.emit(0.5);
-                    port_size.set_target_value(HIGHLIGHT_SIZE);
-                    port_opacity.set_target_value(1.0);
-                });
+                 def _hide_all = hide_all.map(f_!([port_size,port_opacity]{
+                     port_size.set_target_value(0.0);
+                     port_opacity.set_target_value(0.0);
+                 }));
 
-                eval_ view.events.mouse_out ([set_port_sizes_all,set_port_opacity_all] {
-                     set_port_sizes_all.emit(0.0);
-                     set_port_opacity_all.emit(0.0);
-                });
+                // Through the provided ID we can infer whether this port should be highlighted.
+                is_selected      <- activate_ports_with_selected.map(move |id| *id == index);
+                show_normal      <- activate_ports_with_selected.gate_not(&is_selected);
+                show_highlighted <- activate_ports_with_selected.gate(&is_selected);
 
-                eval_ view.events.mouse_down(frp.on_port_mouse_down.emit(index));
+                def _show_highlighted = show_highlighted.map(f_!([port_opacity,port_size]{
+                    port_opacity.set_target_value(1.0);
+                    port_size.set_target_value(HIGHLIGHT_SIZE);
+                }));
+
+                def _show_highlighted = show_normal.map(f_!([port_opacity,port_size]
+                    port_opacity.set_target_value(0.5);
+                    port_size.set_target_value(BASE_SIZE);
+                ));
             }
         }
+
+        // FIXME this is a hack to ensure the ports are invisible at startup.
+        // Right noe we get some of FRP mouse events on startup that leave the
+        // ports visible by default.
+        // Once that is fixed, remove this line.
+        on_hide_delay_finish.emit(());
```

Comment:
We need to fix it. IT sounds strange and dangerous. Please inspect it and write to me what exactly mouse events are propagated during startup.